core/translate.c

```diff
@@ -177,13 +195,18 @@ translate_walk_track(dcontext_t *tdcontext, instr_t *inst, translate_walk_t *wal
     if (walk->in_mangle_region &&
         /* On ARM, we spill registers across an app instr, so go solely on xl8 */
         (IF_X86(!instr_is_our_mangling(inst) ||)
-             instr_get_translation(inst) != walk->translation)) {
+         /* handle adjacent mangle regions */
+         IF_X86(translate_walk_exits_mangling_epilogue(tdcontext, inst, walk) ||)
+         /* Entering the mangling region's epilogue can have different xl8 */
```

Comment:
But then the existing translation check would apply, right?  Do we need these new conditions here?

Reply:
Yes, it fails to recognize adjacent mangling regions otherwise.